ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/TargetDefinitionPersistenceHelper.java

```diff
@@ -87,6 +87,7 @@ public class TargetDefinitionPersistenceHelper {
 	static final String ATTR_INCLUDE_ALL_PLATFORMS = "includeAllPlatforms"; //$NON-NLS-1$
 	static final String ATTR_INCLUDE_SOURCE = "includeSource"; //$NON-NLS-1$
 	static final String ATTR_INCLUDE_CONFIGURE_PHASE = "includeConfigurePhase"; //$NON-NLS-1$
+	static final String ATTR_INCLUDE_REFERENCES = "includeReferences"; //$NON-NLS-1$
```

Comment:
In P2 director application this option is named `followReferences` maybe we should for consitency name this here the same
```suggestion
	static final String ATTR_INCLUDE_REFERENCES = "followReferences"; //$NON-NLS-1$
```

Reply:
I would even go one step further and name it `followRepositoryReferences` otherwise the chances are high that it is confused with something like inclusion of transitive dependencies or alike.

Reply:
I'm not a fan of exaggerated naming schemes. People should not "guess" things and anyone that is uncertain can simply use the UI.

Reply:
> People should not "guess" things

Absolutely, That's why I suggested a more clear name. Of course finding a suitable name is not easy, since it should be as short as possible but also expressive and clear. Yes it's longer, but that does not automatically makes it exaggerated, its always a compromise and one has to find a balance.

> and anyone that is uncertain can simply use the UI.

At least for me the (long-term) goal is actually to require the UI less and since target-definition just like code is read more often than written it should be clear what's intended.

Maybe somebody else also has an opinion?

Reply:
I prefer the longer name that makes it 100% clear that we are talking about repository references. I’ll bet 99% of the users have no clue what that even is so more descriptive is important.

Reply:
`followReferences` is the parameter name of the director and no one ever was "confused" and I'll bett 99% of the people don't even know what repository reference are either... so for these one needs a good UI with a descriptive help if one really is concerned about these 99% of people because longer names won't help them in any ways. We also don't use the term "reference" anywhere else so there is really not any way to be unclear what is meant here.

Reply:
We’re all entitled to opinions and personal preferences.  😜 They need not be the same as someone else’s.  I won’t complain about the decision even if it is different from the one I would make.  Don’t worry, be happy.

Reply:
PDE's and P2's users are not identical also with regard to their experience.

And what's so bad about the extra word?
It's just written if set to false and therefore won't be visible most of the time anyway.

Reply:
This location type is the "P2 Location" it is all about P2, so if it won't be visible most of the time anyway whats wrong with having it consistent with P2 terminology? 😜